Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm

```diff
@@ -394,6 +398,93 @@ - (void)_dismissPopup
 @end
 #endif // PLATFORM(IOS_FAMILY)
 
+#if PLATFORM(MAC)
+@interface _WKWebExtensionActionPopover : NSPopover <NSPopoverDelegate>
+@end
+
+@implementation _WKWebExtensionActionPopover {
+    WeakPtr<WebExtensionAction> _webExtensionAction;
+    _WKWebExtensionActionWebView *_popupWebView;
+}
+
+- (instancetype)initWithWebExtensionAction:(WebExtensionAction&)action
+{
+    if (!(self = [super init]))
+        return nil;
+
+    RefPtr extensionContext = action.extensionContext();
+    if (!extensionContext)
+        return nil;
+
+    _webExtensionAction = action;
+    _popupWebView = dynamic_objc_cast<_WKWebExtensionActionWebView>(action.popupWebView());
+
+    _popupWebView.accessibilityRole = NSAccessibilityPopoverRole;
+
+    [_popupWebView setContentHuggingPriority:NSLayoutPriorityDefaultHigh forOrientation:NSLayoutConstraintOrientationHorizontal];
+    [_popupWebView setContentHuggingPriority:NSLayoutPriorityDefaultHigh forOrientation:NSLayoutConstraintOrientationVertical];
+
+    [NSLayoutConstraint activateConstraints:@[
+        [_popupWebView.widthAnchor constraintGreaterThanOrEqualToConstant:minimumPopoverWidth],
+        [_popupWebView.widthAnchor constraintLessThanOrEqualToConstant:maximumPopoverWidth],
+        [_popupWebView.heightAnchor constraintGreaterThanOrEqualToConstant:minimumPopoverHeight],
+        [_popupWebView.heightAnchor constraintLessThanOrEqualToConstant:maximumPopoverHeight]
+    ]];
+
+    auto *viewController = [[NSViewController alloc] init];
+    viewController.view = _popupWebView;
+    viewController.preferredContentSize = _popupWebView.intrinsicContentSize;
+
+    self.contentViewController = viewController;
+    self.behavior = NSPopoverBehaviorSemitransient;
+
+    action.setPopupPopoverAppearance(action.popupPopoverAppearance());
+
+    [NSNotificationCenter.defaultCenter addObserver:self selector:@selector(_otherPopoverWillShow:) name:NSPopoverWillShowNotification object:nil];
+
+    return self;
+}
+
+- (void)dealloc
+{
+    [NSNotificationCenter.defaultCenter removeObserver:self];
```

Comment:
This isn't necessary, Foundation will do this automatically on all the versions of iOS and macOS we care about.